Adds middleware to override django settings

rehan99000 opened the pull request:
**Description:**
This PR adds the following things:
- Adds a middleware that overrides settings if  `DJANGO_SETTINGS_OVERRIDE` values are present in the site configuration
- Adds tests for it
- Removes unnecessary classes
- Adds middleware to both LMS and Studio

**JIRA:**
https://edlyio.atlassian.net/browse/EDLY-1334

zubair-arbi commented on openedx/features/edly/middleware.py:
```diff
@@ -24,3 +26,31 @@ def process_request(self, request):
+            current_site_configuration = current_site.configuration
+        except SiteConfiguration.DoesNotExist:
+            logger.warning('Site (%s) has no related site configuration.', current_site)
+            return None
```
nit: remove this line

rehan99000 replied:
If I don't add this line, the middleware tries to access values of site configuration that doesn't exist.

zubair-arbi commented on openedx/features/edly/middleware.py:
```diff
@@ -24,3 +26,31 @@ def process_request(self, request):
+            logger.warning('Site (%s) has no related site configuration.', current_site)
+            return None
+
+        if current_site_configuration.values:
```
Just check `if current_site_configuration:`

rehan99000 replied:
It is possible that a site configuration is present but there are no values in it. This is for that case.

zubair-arbi replied:
Wouldn't that be handled by `current_site_configuration.get_value('DJANGO_SETTINGS_OVERRIDE', None)` below?

rehan99000 replied:
no. `get_value` tries to get value from `site_configuration.values` which is `None` in this case so it raises error that can't get from NoneType

zubair-arbi replied:
Ok

ziafazal commented on openedx/features/edly/tests/test_middleware.py:
```diff
@@ -71,3 +74,91 @@ def test_super_user_has_all_sites_access(self):
+                )
+            )
+
+    def test_settings_override_middleware_logs_warning_if_site_configuration_has_no_django_settings_overrides(self):
```
nit: we should avoid such long names for test methods

zubair-arbi replied:
just use `test_settings_override_middleware_logs_warning`

zubair-arbi commented on openedx/features/edly/tests/test_middleware.py:
```diff
@@ -71,3 +74,91 @@ def test_super_user_has_all_sites_access(self):
+
+class SettingsOverrideMiddlewareTests(TestCase):
+    """
+    Tests for "SettingsOverrideMiddleware".
```
`Tests settings override middleware for sites.`

rehan99000 replied:
done

zubair-arbi commented on openedx/features/edly/tests/test_middleware.py:
```diff
@@ -71,3 +74,91 @@ def test_super_user_has_all_sites_access(self):
+    """
+    def setUp(self):
+        """
+        Set up values for tests.
```
`Create environment for settings override middleware tests.`

rehan99000 replied:
done

zubair-arbi commented on openedx/features/edly/tests/test_middleware.py:
```diff
@@ -71,3 +74,91 @@ def test_super_user_has_all_sites_access(self):
+            site=self.request.site,
+            values={
+                'DJANGO_SETTINGS_OVERRIDE': {
+                    'EDLY_USER_INFO_COOKIE_NAME': 'edly-user-info-test'
```
We will never change this cookie name. Use some other variables which we will override in LMS, CMS (https://docs.google.com/spreadsheets/d/19smxeJAv4T3wr6KLfEXxkZP6_Ay0yiAFRETXVBRv9eQ/edit#gid=454576137)

Please use following list of variable to override (in same order)

`SITE_NAME`, 
`CMS_BASE`, 
`LMS_BASE`, 
`LMS_ROOT_URL`, 
`SESSION_COOKIE_DOMAIN`, 
`LOGIN_REDIRECT_WHITELIST`, 
`DEFAULT_FEEDBACK_EMAIL`,
`CREDENTIALS_PUBLIC_SERVICE_URL`

rehan99000 replied:
done

zubair-arbi commented on openedx/features/edly/tests/test_middleware.py:
```diff
@@ -71,3 +74,91 @@ def test_super_user_has_all_sites_access(self):
+        )
+        assert settings.EDLY_USER_INFO_COOKIE_NAME == 'edly-user-info'
+        self.client.get('/', follow=True)
+        assert settings.EDLY_USER_INFO_COOKIE_NAME == 'edly-user-info-test'
```
Since we will need to validate multiple values so create a helper method (https://github.com/edly-io/edly-panel-backend/pull/6/files):
```python
def _assert_response_settings_values(self, response, expected_site_configuration_values):
    """
    Validate configuration settings in HTTP response.
    """
    for config_key, expected_config_value in expected_site_configuration_values.items():
        assert expected_config_value == getattr(response.context['settings'], config_key)
```

rehan99000 replied:
done

rehan99000 commented:
I have added test to check for the disabled site configuration. I have also added a helper method to validate current settings against expected settings values passed in the parameter.

zubair-arbi commented on openedx/features/edly/tests/test_middleware.py:
```diff
@@ -71,3 +75,137 @@ def test_super_user_has_all_sites_access(self):
+            ]
+        }
+
+    def _assert_settings_equal(self, expected_settings_values):
```
```python
def _assert_response_settings_values(self, response, expected_site_configuration_values):
    """
    Validate configuration settings in HTTP response.
    """
    for config_key, expected_config_value in expected_site_configuration_values.items():
        assert expected_config_value == getattr(response.context['settings'], config_key)
```

rehan99000 replied:
these settings are django settings not available in response. that's why i set this name

rehan99000 commented on openedx/features/edly/tests/test_middleware.py:
```diff
@@ -71,3 +75,137 @@ def test_super_user_has_all_sites_access(self):
+        SiteConfigurationFactory(
+            site=self.request.site,
+            values={
+                'PLATFORM_NAME': 'St. NormanX',
```
done

rehan99000 commented on openedx/features/edly/tests/test_middleware.py:
```diff
@@ -71,3 +75,137 @@ def test_super_user_has_all_sites_access(self):
+            'SITE_NAME': '',
+            'CMS_BASE': 'localhost:8010',
+            'LMS_BASE': 'localhost:8000',
+            'LMS_ROOT_URL': 'edx.devstack.lms',
```
done
